From 2c453e81b78c46460783316f152c90b45570ec84 Mon Sep 17 00:00:00 2001
From: croniccorey <cronmod.dev@gmail.com>
Date: Wed, 24 Jun 2015 18:07:18 -0400
Subject: [PATCH 34/61] [AMLUtils] Add support for framerate automation

 This fixs an issue resulting in loss of HDMI signal during playback of some videos

Credit: svlasic@gmail.com
---
 xbmc/utils/AMLUtils.cpp |   30 ++++++++++++++++++++++++------
 1 file changed, 24 insertions(+), 6 deletions(-)

diff --git a/xbmc/utils/AMLUtils.cpp b/xbmc/utils/AMLUtils.cpp
index 72dfb28..ce9d07e 100644
--- a/xbmc/utils/AMLUtils.cpp
+++ b/xbmc/utils/AMLUtils.cpp
@@ -344,7 +344,25 @@ bool aml_mode_to_resolution(const char *mode, RESOLUTION_INFO *res)
     res->fRefreshRate = 60;
     res->dwFlags = D3DPRESENTFLAG_PROGRESSIVE;
   }
-  else if (fromMode.Equals("720p"))
+  else if (fromMode.Equals("480cvbs"))
+  {
+    res->iWidth = 720;
+    res->iHeight= 480;
+    res->iScreenWidth = 720;
+    res->iScreenHeight= 480;
+    res->fRefreshRate = 60;
+    res->dwFlags = D3DPRESENTFLAG_INTERLACED;
+  }
+  else if (fromMode.Equals("576cvbs"))
+  {
+    res->iWidth = 720;
+    res->iHeight= 576;
+    res->iScreenWidth = 720;
+    res->iScreenHeight= 576;
+    res->fRefreshRate = 50;
+    res->dwFlags = D3DPRESENTFLAG_INTERLACED;
+  }
+  else if (fromMode.Equals("720p") || fromMode.Equals("720p59hz") || fromMode.Equals("720p60hz"))
   {
     res->iWidth = 1280;
     res->iHeight= 720;
@@ -362,7 +380,7 @@ bool aml_mode_to_resolution(const char *mode, RESOLUTION_INFO *res)
     res->fRefreshRate = 50;
     res->dwFlags = D3DPRESENTFLAG_PROGRESSIVE;
   }
-  else if (fromMode.Equals("1080p"))
+  else if (fromMode.Equals("1080p") || fromMode.Equals("1080p59hz") || fromMode.Equals("1080p60hz"))
   {
     res->iWidth = 1920;
     res->iHeight= 1080;
@@ -371,7 +389,7 @@ bool aml_mode_to_resolution(const char *mode, RESOLUTION_INFO *res)
     res->fRefreshRate = 60;
     res->dwFlags = D3DPRESENTFLAG_PROGRESSIVE;
   }
-  else if (fromMode.Equals("1080p24hz"))
+  else if (fromMode.Equals("1080p23hz") || fromMode.Equals("1080p24hz"))
   {
     res->iWidth = 1920;
     res->iHeight= 1080;
@@ -398,7 +416,7 @@ bool aml_mode_to_resolution(const char *mode, RESOLUTION_INFO *res)
     res->fRefreshRate = 50;
     res->dwFlags = D3DPRESENTFLAG_PROGRESSIVE;
   }
-  else if (fromMode.Equals("1080i"))
+  else if (fromMode.Equals("1080i") || fromMode.Equals("1080i59hz") || fromMode.Equals("1080i60hz"))
   {
     res->iWidth = 1920;
     res->iHeight= 1080;
@@ -425,7 +443,7 @@ bool aml_mode_to_resolution(const char *mode, RESOLUTION_INFO *res)
     res->fRefreshRate = 24;
     res->dwFlags = D3DPRESENTFLAG_PROGRESSIVE;
   }
-  else if (fromMode.Equals("4k2k24hz"))
+  else if (fromMode.Equals("4k2k23hz") || fromMode.Equals("4k2k24hz"))
   {
     res->iWidth = 1920;
     res->iHeight= 1080;
@@ -443,7 +461,7 @@ bool aml_mode_to_resolution(const char *mode, RESOLUTION_INFO *res)
     res->fRefreshRate = 25;
     res->dwFlags = D3DPRESENTFLAG_PROGRESSIVE;
   }
-  else if (fromMode.Equals("4k2k30hz"))
+  else if (fromMode.Equals("4k2k29hz") || fromMode.Equals("4k2k30hz"))
   {
     res->iWidth = 1920;
     res->iHeight= 1080;
-- 
1.7.10.4

